File: py/solns/longestPalindromicSubstr/longestPalindromicSubstr.py
class Solution:
    def __init__(self):
        self.res = ""
        self.maxLen = 0
    def naive(self,s):
        self.length = len(s)
        def loop(start,end):
            l,r = start,end
            while l>=0 and r<=self.length-1 and s[l]==s[r]:
                if r-l+1>self.maxLen:
                    self.res = s[l:r+1]
                    self.maxLen = r-l+1
                l-=1
                r+=1
        for i in range(self.length):
            loop(i,i)
            loop(i,i+1)
        return self.res

File: cpp/example/src/LongestPalindromicSubstr/longestPalindromicSubstr.cpp
#include "longestPalindromicSubstr.h"

// int longestPalindromicSubstr::naive(int n){
//     return n;
// }

longestPalindromicSubstr::longestPalindromicSubstr(){
    this->res = "";
    this->maxLen = 0;
};
void longestPalindromicSubstr::loop(int start,int end){
    int l=start,r=end;
    while (l>=0 && r<=length-1 && this->s[l]==this->s[r]){
        if (r-l+1>this->maxLen){
            this->maxLen = r-l+1;
            this->res = this->s.substr(l,r-l+1);
        }
        l-=1;
        r+=1;
    }
};

string longestPalindromicSubstr::naive(string s){
    this->s = s;
    this->length = s.size();
    for (int i=0;i<length;i++){
        loop(i,i);
        loop(i,i+1);
    }
    return this->res;
};